malcolm/core/models.py

```diff
@@ -125,6 +126,62 @@ def apply_change(self, path, *args):
                 "Cannot process change %s" % ([self.path + path] + list(args))
             getattr(self, "set_%s" % path[0])(args[0])
 
+with Anno("Lower limit"):
+    ALoLimit = np.float64
+with Anno("Upper limit"):
+    AHiLimit = np.float64
+with Anno("Description"):
+    ADescription = str
+with Anno("Number of significant figures to display"):
+    APrecision = np.int32
+with Anno("format to display value"):
+    AForm = str
+with Anno("Units"):
+    AUnits = str
+
+@Serializable.register_subclass("display_t")
+class Display(Model):
+
+    __slots__ = ["limitLow", "limitHigh", "description", "precision", "form", "units"]
+
+    # noinspection PyPep8Naming
+    # limitLow and limitHigh are camelCase to maintain compatibility with
+    # EPICS normative types
+    def __init__(self, limitLow=0, limitHigh=0, description="", precision=0, form=None, units=None):
+        # type: (ALoLimit, AHiLimit, ADescription, APrecision, AForm, AUnits) -> None
+        # Set initial values
+        self.limitLow = self.set_limitLow(limitLow)
+        self.limitHigh = self.set_limitHigh(limitHigh)
+        self.description = self.set_description(description)
+        self.form = self.set_form(form)
+        self.precision = self.set_precision(precision)
+        self.units = self.set_units(units)
+
+    def set_limitLow(self, limitLow):
+        return self.set_endpoint_data("limitLow", np.float64(limitLow))
```

Comment:
no need to cast here, or in any set_ function, set_endpoint_data will do this for us